### PR TITLE
Complete test implementation

### LINKED ISSUE #1118: Complete prover_skips_preproven_commitments_test

### PR DESCRIPTION
**Pull request by rakanalh**
## Linked Issues
- Fixes #1118

### CODE REVIEW

File: crates/prover/src/da_block_handler.rs [hunk 1]
```diff
@@ -182,6 +182,10 @@ where
                 l1_block.header().height(),
             );
 
+            // Make sure all sequencer commitments are stored in ascending order.
+            // We sort before checking ranges to prevent substraction errors.
+            sequencer_commitments.sort();
```
Comment (eyusufatik) [inline crates/prover/src/da_block_handler.rs:187]: remove sort below?

**Review by eyusufatik — CHANGES_REQUESTED** [review]
small nit, looks nice

File: crates/prover/src/da_block_handler.rs [hunk 2]
```diff
@@ -182,6 +182,10 @@ where
...
+
```
Comment (rakanalh) [inline crates/prover/src/da_block_handler.rs:188]:
```suggestion
```

Comment (eyusufatik) [reply crates/prover/src/da_block_handler.rs]: can you add those comments on the new sort?

File: crates/prover/src/da_block_handler.rs (hunk 1 above)
Comment (rakanalh) [reply crates/prover/src/da_block_handler.rs:187]: Removed

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**